### PR TITLE
CHANGE @W-16891041@ FlowTest rules consolidated

### PR DESCRIPTION
**Pull request by jfeingold35**

### CODE REVIEW

File: packages/code-analyzer-flowtest-engine/src/engine.ts [hunk 1]
```diff
@@ -51,9 +51,9 @@ export class FlowTestEngine extends Engine {
                 return [];
             }
         }
-        const flowTestRules: FlowTestRuleDescriptor[] = await this.commandWrapper.getFlowTestRuleDescriptions();
         this.emitDescribeRulesProgressEvent(75);
-        const convertedRules = flowTestRules.map(toRuleDescription);
+        const consolidatedNames: Set<string> = new Set(getConsolidatedRuleNames());
+        const convertedRules: RuleDescription[] = [...consolidatedNames.values()].map(getConsolidatedRuleByName);
```
Comment (stephen-carter-at-sf) [inline packages/code-analyzer-flowtest-engine/src/engine.ts]: Why use a set? Why not just do `getConsolidatedRuleNames().map(getConsolidatedRuleByName)`?

File: packages/code-analyzer-flowtest-engine/src/engine.ts [hunk 2]
```diff
@@ -15,6 +14,7 @@ import {
     FlowTestExecutionResult,
     FlowTestRuleDescriptor
```
Comment (stephen-carter-at-sf) [inline packages/code-analyzer-flowtest-engine/src/engine.ts]: Is this needed?

File: packages/code-analyzer-flowtest-engine/src/engine.ts
Comment (stephen-carter-at-sf) [inline packages/code-analyzer-flowtest-engine/src/engine.ts:1]: It looks like we have tabs mixed with spaces. Can you make it all spaces?

Comment (jfeingold35) [reply packages/code-analyzer-flowtest-engine/src/engine.ts:1]: Where am I using tabs?

File: packages/code-analyzer-flowtest-engine/src/hardcoded-catalog.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        description: getMessage('ConsolidatedRuleDescription', 'With Sharing'),
+        severityLevel: SeverityLevel.Low,
+        tags: ['Recommended', 'Security'],
+        resourceUrls: []
+    }]
+]);
+
+export function getConsolidatedRuleNames(): string[] {
+    return [...QUERY_NAMES_TO_CONSOLIDATED_NAMES.values()];
+}
```
Comment (stephen-carter-at-sf) [inline packages/code-analyzer-flowtest-engine/src/hardcoded-catalog.ts]: Oh I see... you use a set in the other spot. I think it should be here to make things unique. So you could here do
`[...new Set(QUERY_NAMES_TO_CONSOLIDATED_NAMES.values())]`

**Review by stephen-carter-at-sf — APPROVED** [review]
Consider moving the location where you make the consolidatedRuleNames unique (i.e. apply the set inside of the getConsolidatedRuleNames function instead. 

Otherwise, approve.

**Pull request merged**